crawl-ref/source/beam.cc

```diff
@@ -4623,74 +4624,17 @@ void bolt::knockback_actor(actor *act, int dam)
     const int roll = origin_spell == SPELL_FORCE_LANCE
                      ? 7 + 0.27 * ench_power
                      : 17;
-    const int weight = max_corpse_chunks(act->is_monster() ? act->type :
-                                                            you.mons_species());
 
-    const coord_def oldpos = act->pos();
-
-    if (act->is_stationary())
-        return;
-    // We can't do knockback if the beam starts and ends on the same space
-    if (source == oldpos)
-        return;
-    ASSERT(ray.pos() == oldpos);
-
-    coord_def newpos = oldpos;
-    for (int dist_travelled = 0; dist_travelled < distance; ++dist_travelled)
-    {
-        if (x_chance_in_y(weight, roll))
-            continue;
-
-        const ray_def oldray(ray);
-
-        ray.advance();
-
-        newpos = ray.pos();
-        if (newpos == oldray.pos()
-            || cell_is_solid(newpos)
-            || actor_at(newpos)
-            || !act->can_pass_through(newpos)
-            || !act->is_habitable(newpos))
-        {
-            ray = oldray;
-            break;
-        }
+    const char* message = origin_spell == SPELL_CHILLING_BREATH
+                ? make_stringf("%s %s blown backwards by the freezing wind.",
+                    act->name(DESC_THE).c_str(),
+                    act->conj_verb("are").c_str()).c_str()
+                : make_stringf("%s %s knocked back by the %s.",
+                    act->name(DESC_THE).c_str(),
+                    act->conj_verb("are").c_str(),
+                    name.c_str()).c_str();
```

Comment:
Let's simplify this to `%s %s knocked back by the %s` and only pass the last element ("freezing wind" / beam.name) into `knockback()`.